### PR TITLE
Hosting: Add a new Miscellaneous Card for clearing cache

### PR DESCRIPTION
**Pull request by aidvu**
#### Changes proposed in this Pull Request

We want other admin stuff to live here.
One example is the Clear Cache action.

We want to allow users to be able to clear cache
if an HE has enabled this ability for them.

#### Testing instructions

Calypso:
* Create an Atomic site
* Go to Hosting Configuration
* See Miscellaneous Card
* See Clear Cache button
* Click it
* Enter reason
* Check logstash for logs

#### New card looks

Screenshot from 2020-04-16 15-04-23

and the reason popup.

Screenshot from 2020-04-16 15-04-52

### CODE REVIEW

File: client/my-sites/hosting/miscellaneous-card/index.js [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+	const closeConfirmationDialog = () => {
+		setShowDialog( false );
+	};
+
+	const onReasonChange = event => {
+		setReason( event.target.value );
+	};
+
+	const getClearCacheContent = () => {
+		const clearCacheText = translate( 'Clear Cache' );
```
Comment (a8ci18n) [inline client/my-sites/hosting/miscellaneous-card/index.js:55]: :information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **9** times:
`translate( 'Clear cache on error.' )` *ES Score: 7*

Comment (aidvu) [reply client/my-sites/hosting/miscellaneous-card/index.js:55]: Not good for our use case.

File: client/my-sites/hosting/miscellaneous-card/index.js [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+					{ translate( '{{strong}}Warning!{{/strong}}', {
+						components: {
+							strong: <strong />,
+						},
+					} ) }
+				</p>
+				<p>
+					{ translate(
+						'Although we allow WordPress cache clearing, it might make your site unresponsive or even ' +
+							'break it while the cache is rebuilding.'
```
Comment (jmdodd) [inline client/my-sites/hosting/miscellaneous-card/index.js:76]: Clearing the cache on your site may make it unresponsive while the cache is rebuilding. Please use this feature responsibly.

File: client/my-sites/hosting/miscellaneous-card/index.js [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+					isVisible={ showDialog }
+					buttons={ deleteButtons }
+					className="miscellaneous-card__confirm-dialog"
+				>
+					<h1 className="miscellaneous-card__confirm-header">{ clearCacheText }</h1>
+					<FormLabel
+						htmlFor="confirmDomainChangeInput"
+						className="miscellaneous-card__confirm-label"
+					>
+						<p>{ translate( 'Please tell us why you think a cache clear is required.' ) }</p>
```
Comment (jmdodd) [inline client/my-sites/hosting/miscellaneous-card/index.js:97]: Please let us know why you are clearing your site's cache.

**Pull request merged**